### PR TITLE
A new hope

### PR DESCRIPTION
**Pull request by AndrewEQ**
- added fresh paint
- spaced the menu items for better readibility
- updated the title
- updated the syntax highlighter (part of fresh paint update)
- added sticky menu feature
- added callout feature
- updated currently affected articles with callout feature

Callout usage:
"> [callout_warn] My sentence"
"> [callout_success] My sentence"
"> [callout_danger] My sentence"
"> [callout_info] My sentence"
"> [callout_light] My sentence"
"> [callout_dark] My sentence"

### CODE REVIEW

**Comment by kusma** [comment]
@AndrewEQ: Could you separate out the different bulleted items into separate commits?

**Comment by AndrewEQ** [comment]
Your wish is my command 😄 ...

**Comment by kusma** [comment]
Hmm... I'm trying to figure out how to preview the end-result...

**Comment by AndrewEQ** [comment]
Was wondering how the original guys tested it, please let me know if you figure it out; I had to use the web inspector and online environments for the component.

You could try it out and revert when done or branch a test version?

**Comment by kusma** [comment]
I think I used to have a Docker image that built this or something... it was done in a hurry to make things public before I left the company, so not everything was done perfectly, sadly.

**Comment by kusma** [comment]
Ideally I wanted to migrate the docs to something like [readthedocs](https://readthedocs.org/), but I never found the time for that... In particular, I had no idea how to deal with the API reference docs in that case.

**Comment by AndrewEQ** [comment]
Yeah, that would be cool, hmm: https://docs.readthedocs.io/en/latest/intro/import-guide.html

**Comment by AndrewEQ** [comment]
We'd have to prototype it sometime

**Comment by AndrewEQ** [comment]
So what's our current process for the docs or API references to be generated?

**Comment by AndrewEQ** [comment]
This one also looks nice: https://docsify.js.org

**Comment by LuisRodriguezLD** [comment]
I personally like [docusaurus.io](https://docusaurus.io). Just my two cents.

**Comment by AndrewEQ** [comment]
Hmm, yeah, actually its also pretty clean and has the extra sub-menu on the right.

**Comment by kusma** [comment]
OK, so I managed to test this out, and sadly there's a few problems.

1. The scroll-bars for the menu is misplaced, and sticks a bit into the menu-area.
2. The small-size version are, uh, not at all working. The menu starts overlapping the text pretty quickly, and at the smallest breakpoints they're completely overlapping.
3. Syntax highlighting seems to be broken.

That being said, I really like the callout-coloring etc. But are they supposed to contain the "[callout-info]" tags in the text? I know they already do, but perhaps we could remove it now?

File: assets/prism/prism.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
...@("|')(?:\1\1|\\[\s\S]|(?!\1)[^\\])*\1/,greedy:!0},{pattern:/("|')(?:\\.|(?!\1)[^\\\r\n])*?\1/,greedy:!0}],"class-name":[{pattern:/\b[A-Z]\w*(?:\.\w+)*\b(?=\s+\w+)/,inside:{punctuation:/\./}},{pattern:/(\[)[A-Z]\w*(?:\.\w+)*\b/,lookbehind:!0,inside:{punctuation:/\./}},{pattern:/(\b(?:class|interface)\s+[A-Z]\w*(?:\.\w+)*\s*:\s*)[A-Z]\w*(?:\.\w+)*\b/,lookbehind:!0,inside:{punctuation:/\./}},{pattern:/((?:\b(?:class|interface|new)\s+)|(?:catch\s+\())[A-Z]\w*(?:\.\w+)*\b/,lookbehind:!0,inside:{punctuation:/\./}}],number:/\b0x[\da-f]+\b|(?:\b\d+\.?\d*|\B\.\d+)f?/i,operator:/>>=?|<<=?|[-=]>|([-+&|?])\1|~|[-+*/%&|^!=<>]=?/,punctuation:/\?\.?|::|[{}[\];(),.:]/}),Prism.languages.insertBefore("csharp","class-name",{"generic-method":{pattern:/\w+\s*<[^>\r\n]+?>\s*(?=\()/,inside:{function:/^\w+/,"class-name":{pattern:/\b[A-Z]\w*(?:\.\w+)*\b/,inside:{punctuation:/\./}},keyword:Prism.languages.csharp.keyword,punctuation:/[<>(),.:]/}},preprocessor:{pattern:/(^\s*)#.*/m,lookbehind:!0,alias:"property",inside:{directive:{pattern:/(\s*#)\b(?:define|elif|else|endif|endregion|error|if|line|pragma|region|undef|warning)\b/,lookbehind:!0,alias:"keyword"}}}}),Prism.languages.dotnet=Prism.languages.csharp;
```
Comment (kusma) [inline assets/prism/prism.js]: Strange. The code is different if I download it myself...

File: assets/site.js [hunk 2]
```diff
@@ -29,4 +29,19 @@ $(document).ready(function() {
...
+    $("blockquote:contains('[callout_success]')").addClass("callout-success");
+    $("blockquote:contains('[callout_success]')").html($("blockquote:contains('[callout_success]')").html().replace('[callout_success]',''));
+    $("blockquote:contains('[callout_danger]')").addClass("callout-danger");
+    $("blockquote:contains('[callout_danger]')").html($("blockquote:contains('[callout_danger]')").html().replace('[callout_danger]',''));
+    $("blockquote:contains('[callout_info]')").addClass("callout-info");
+    $("blockquote:contains('[callout_info]')").html($("blockquote:contains('[callout_info]')").html().replace('[callout_info]',''));
+    $("blockquote:contains('[callout_light]')").addClass("callout-light");
+    $("blockquote:contains('[callout_light]')").html($("blockquote:contains('[callout_light]')").html().replace('[callout_light]',''));
+    $("blockquote:contains('[callout_dark]')").addClass("callout-dark");
+    $("blockquote:contains('[callout_dark]')").html($("blockquote:contains('[callout_dark]')").html().replace('[callout_dark]',''));
```
Comment (kusma) [inline assets/site.js]: I think it would be simpler and cleaner to just use inline HTML in the form
```markdown
<blockquote class="callout-warn">
Foo bar
<blockquote>
```
... rather than having to match things with JS client-side. If you agree, I have an alternative branch for that part, that I can send as a PR credited in your name...

File: assets/site.css [hunk 3]
```diff
@@ -40,6 +40,20 @@ a:hover {
...
+.sidebar-sticky {
+    height: 100%;
+    width: 200px;
+    position: fixed;
+    z-index: 900;
+    top: 1rem;
+    overflow-x: hidden;
+    padding-top: 1rem;
+    padding-bottom: 1rem;
+}
```
Comment (kusma) [inline assets/site.css]: Heh, seems we're already using this non-existent class in layout.html... I guess this is left-over from some previous version...

File: layout.html [hunk 4]
```diff
@@ -16,7 +16,7 @@
   <body>
 
     <nav class="navbar navbar-dark sticky-top bg-dark flex-md-nowrap p-0">
-      <a class="navbar-brand col-sm-3 col-md-2 mr-0" href="##BASE_URL##">Fuse: Documentation</a>
+      <a class="navbar-brand col-sm-3 col-md-2 mr-0" href="##BASE_URL##">Fuse Open Documentation</a>
```
Comment (kusma) [inline layout.html]: I think this perhaps deserves its own commit. Nice catch!

File: outline [hunk 5]
```diff
@@ -185,10 +188,6 @@ Fuse : articles/index.md
 		Starting an Android Activity : articles/native-interop/android-start-activity.md
 		iOS Capabilities : articles/native-interop/ios-capabilities.md
 
-	Alive UX Kit : articles/alive/alive.md
-		Components : api-docs/api/alive.json
-		Themes : articles/alive/themes.md
-
```
Comment (kusma) [inline outline]: How did you update this file? By hand, or is there some automatic way?

File: outline [hunk 6]
```diff
@@ -111,7 +111,7 @@ Fuse : articles/index.md
 		Path : api-docs/api/fuse/controls/path.json
 		Image : api-docs/api/fuse/controls/image.json
 		Video : api-docs/api/fuse/controls/video.json
-		// Where is this located? Sound : api-docs/api/fuse/controls/sound.json
+		Sound : api-docs/api/fuse/playsound.json
```
Comment (kusma) [inline outline]: Nice!

File: assets/prism/prism.js (hunk 1 above)
Comment (AndrewEQ) [reply assets/prism/prism.js]: I dunno why its breaking but the main difference with prism was using the Okaidia theme instead of the default; suits our new color scheme better.

File: assets/site.js (hunk 2 above)
Comment (AndrewEQ) [reply assets/site.js]: Yeah man, I don't mind, its only a little more typing, I was just trying to make the previous guys idea work, lol.

File: assets/site.css (hunk 3 above)
Comment (AndrewEQ) [reply assets/site.css]: The class was there before, and its a good idea, so I tried to fulfil the functionality.

File: layout.html (hunk 4 above)
Comment (AndrewEQ) [reply layout.html]: Ok

File: assets/prism/prism.js (hunk 1 above)
Comment (kusma) [reply assets/prism/prism.js]: Yeah, but I did that as well, and still got a different result. It's odd...

File: assets/site.js (hunk 2 above)
Comment (kusma) [reply assets/site.js]: Yeah, but I think extending markdown is a bit of a silly thing to do when we can just inline html. But OK, I'll submit my version, then!

File: outline (hunk 5 above)
Comment (AndrewEQ) [reply outline]: I did it by hand, mostly cutting and pasting though to retain previous formatting. Only real new things were around Observables, Models API and sound.

File: outline (hunk 6 above)
Comment (AndrewEQ) [reply outline]: So it worked! 😄

File: assets/site.css (hunk 3 above)
Comment (kusma) [reply assets/site.css]: Yeah, I get that. I was just puzzled that we're using an undefined class. Do you know if any linters pick up things like this?

File: assets/prism/prism.js (hunk 1 above)
Comment (AndrewEQ) [reply assets/prism/prism.js]: Oh, yeah, that is odd, perhaps different browser or platform difference?

Comment (AndrewEQ) [reply assets/prism/prism.js]: Lets go with the one that works haha

File: assets/site.js (hunk 2 above)
Comment (AndrewEQ) [reply assets/site.js]: 👍

File: outline (hunk 5 above)
Comment (kusma) [reply outline]: OK, thanks :)

File: outline (hunk 6 above)
Comment (kusma) [reply outline]: Oh, I'm not sure I actually tested it! I'll check that next time I find some extra time for this :)

File: assets/site.css (hunk 3 above)
Comment (AndrewEQ) [reply assets/site.css]: Funny that you ask, I've built a tool 😄  that can remove unused CSS by comparing the input CSS with the CSS used in HTML: https://www.npmjs.com/package/css-purge

File: outline (hunk 5 above)
Comment (AndrewEQ) [reply outline]: Oh I did reduce the one redundant part, which was the "Full Reference" link and sub links.

File: outline (hunk 6 above)
Comment (AndrewEQ) [reply outline]: Heehee, cool

**Comment by AndrewEQ** [comment]
I shall work on the menu a bit more in the meantime

**Comment by kusma** [comment]
I think we landed all of the important changes from this one, no? If not, please rebase and reopen :)
